### PR TITLE
feat(ui-markdown-editor): init thematic break functionality - I24

### LINKED ISSUE #24: User interface to insert a thematic break
Markdown supports the concept of a _thematic break_ which markdown-pdf interprets as a page break (`---` in markdown). Having a toolbar item to insert a thematic break in the markdown editor would allow users to control where page breaks occur in PDFs that are generated from the contents of the editor.

The thematic break should be rendered in Slate to indicate that a page break will occur. E.g.

```
---- page break ---- 
```

Or similar.

In CommonMark this is represented by an `org.accordproject.commonmark.ThematicBreak` node. See: 
https://models.accordproject.org/markdown/commonmark@0.2.0.html

### PR DESCRIPTION
**Pull request by irmerk**
# Closes #24
Initialize UI to support thematic breaks

### Changes
-  Adjust `FormatButton` functionality for inserting `horizontal_rule` void

### Flags
- Needs keyboard shortcut (@Michael-Grover)
- Needs icon for button (@Michael-Grover)
- Needs toolbar placement (@Michael-Grover)
- Strangely, `insertThematicBreak` does not work when `children: [text]` is replaced with `children: [{ text: '' }]`

### Author Checklist
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [x] Extend the documentation, if necessary
- [x] Merging to `master` from `fork:branchname`

### CODE REVIEW

**Comment by Michael-Grover** [comment]
icon:
[page-break.zip](https://github.com/accordproject/web-components/files/4898663/page-break.zip)

shortcut: cmd/ctrl+enter

location:
image
last item

**Comment by Michael-Grover** [comment]
Page break indicator in contract mockup

https://invis.io/9GRP1QB8D2U#/424658590_page_Break

image

**Comment by Michael-Grover** [comment]
sorry for the shortcut I mean cmd+enter on mac and ctrl+enter on Windows

**Comment by Michael-Grover** [comment]
[page break.svg.zip](https://github.com/accordproject/web-components/files/4899310/page.break.svg.zip)

the icon is masked in an awkward way so I added space around the SVG, attached above. If this file still doesn't work let me know
image
image

**Comment by Michael-Grover** [comment]
https://www.loom.com/share/2abe47f40d904f8cb37f97c619e60b4f

encountering this bug when trying to delete a page break by highlighting it and pressing backspace
```Cannot resolve a DOM point from Slate point: {"path":[4,0],"offset":1}
Error: Cannot resolve a DOM point from Slate point: {"path":[4,0],"offset":1}
    at Object.toDOMPoint (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:218262)
    at Object.toDOMRange (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:218492)
    at https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:239335
    at Ii (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:5640772)
    at Ji (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:5640875)
    at Sj (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:5658687)
    at exports.unstable_runWithPriority (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:5678250)
    at cg (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:5602728)
    at Jj (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:5655189)
    at yj (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:5651401)```

**Comment by irmerk** [comment]
> https://www.loom.com/share/2abe47f40d904f8cb37f97c619e60b4f
> 
> encountering this error message when trying to delete a page break by highlighting it and pressing backspace
> 
> ```
> Error: Cannot resolve a DOM point from Slate point: {"path":[4,0],"offset":1}
>     at Object.toDOMPoint (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:218262)
>     at Object.toDOMRange (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:218492)
>     at https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:239335
>     at Ii (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:5640772)
>     at Ji (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:5640875)
>     at Sj (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:5658687)
>     at exports.unstable_runWithPriority (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:5678250)
>     at cg (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:5602728)
>     at Jj (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:5655189)
>     at yj (https://deploy-preview-149--ap-web-components.netlify.app/vendors~main.ee66d2b2e52d3e507292.bundle.js:2:5651401)```
> ```

I'm not able to replicate this.

File: packages/ui-markdown-editor/src/lib/components/Span/HorizontalRule.js [hunk 1]
```diff
@@ -1,12 +1,58 @@
...
+  color: #949CA2;
+  font-family: inherit;
+  font-size: 13px;
+  font-weight: bold;
+  letter-spacing: 0.3px;
+  line-height: 14px;
+`;
+
+const HorizontalRule = (props) => (
+  <div contentEditable={false} {...props.attributes}>
```
Comment (DianaLease) [inline packages/ui-markdown-editor/src/lib/components/Span/HorizontalRule.js:42]: In regards to the error @Michael-Grover saw, maybe adding `user-select: none` in the style of this div would prevent it? according to this issue: https://github.com/ianstormtaylor/slate/issues/3421

**Comment by DianaLease** [comment]
> I can't seem to insert a page break using the keyboard shortcut, but maybe I am doing something wrong. Are others able to?

I think it needs to be added to the hotkeys file and then implemented here: https://github.com/accordproject/web-components/blob/dc4fb5258efba720114a64b3c332114d650f356a/packages/ui-markdown-editor/src/lib/index.js#L58

**Comment by irmerk** [comment]
> > I can't seem to insert a page break using the keyboard shortcut, but maybe I am doing something wrong. Are others able to?
> 
> I think it needs to be added to the hotkeys file and then implemented here:
> 
> https://github.com/accordproject/web-components/blob/dc4fb5258efba720114a64b3c332114d650f356a/packages/ui-markdown-editor/src/lib/index.js#L58

Yeah just adding that now 🤦‍♀️

**Comment by Michael-Grover** [comment]
image
can add outside of a variable too

**Pull request merged**